src/ui/pages/MatchPage/MatchPage.tsx

```diff
@@ -1,9 +1,99 @@
+import CircularProgress from '@mui/material/CircularProgress'
+import Box from '@mui/material/Box'
+import { Suspense } from 'react'
+import { v4 as uuid } from 'uuid'
+
+import { CardInstance, IPlayerSeed } from '../../../game/types'
+import { StorageService, storage } from '../../../services/StorageService'
+import { stubDeck } from '../../../test-utils/stubs/deck'
 import { Match } from '../../components/Match'
 
-import { MatchPageProps } from './types'
+// Simple Suspense resource implementation
+let deckCache: {
+  read: () => CardInstance[]
+} | null = null
+
+const createDeckResource = () => {
+  let status: 'pending' | 'error' | 'success' = 'pending'
+  let result: CardInstance[]
+  let error: unknown
+
+  const suspender = (async () => {
+    try {
+      const savedDeck = await storage.loadDeck()
+
+      if (savedDeck) {
+        result = StorageService.hydrateDeserializedDeck(savedDeck)
+      } else {
+        result = stubDeck()
+      }
+      status = 'success'
+    } catch (e) {
+      error = e
+      status = 'error'
+    }
+  })()
+
+  return {
+    read() {
+      if (status === 'pending') {
+        // eslint-disable-next-line @typescript-eslint/only-throw-error
+        throw suspender
+      } else if (status === 'error') {
+        throw error
+      } else {
+        return result
+      }
+    },
+  }
+}
+
+const getDeckResource = () => {
+  if (!deckCache) {
+    deckCache = createDeckResource()
+  }
+  return deckCache
+}
```

Comment:
@jules it seems that the cached deck data is global and will persist across `MatchPageContent` instances. Each `MatchPageContent` instance must manage its own deck cache state, as the underlying data may change between instances.